### PR TITLE
Allow multiple groups

### PR DESCRIPTION
**Pull request by KhadijaMahanga**
## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [ ]  commits are clean
- [ ]  commit messages are clean

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

File: wazimap_ng/datasets/tasks/indicator_data_extraction.py [hunk 1]
```diff
@@ -17,27 +19,28 @@ def __init__(self, geography_id):
...
-                if len(values) > 1:
-                    logger.warn(f"Expected a single group when creating a subindicator - found {len(values)}!")
-                subindicator = values[0]
-                subindicators[subindicator] = count
+            values = list(datum.values()) 
+            values_length = len(values)
+            if values_length > 0:
+                subindicators = defaultdict(dict)
+                for x in range (values_length -1, -1, -1):
+                    if x == values_length -1:
```
Comment (kilemensi) [inline wazimap_ng/datasets/tasks/indicator_data_extraction.py:36]:
```suggestion
                for x in range (values_length - 1, -1, -1):
                    if x == values_length - 1:
```

File: wazimap_ng/datasets/tasks/indicator_data_extraction.py [hunk 2]
```diff
@@ -66,15 +69,14 @@ def add_subindicator(self, data_blob):
 @transaction.atomic
 def indicator_data_extraction(indicator, **kwargs):
     sorter = Sorter()
-    primary_group = indicator.groups[0] # TODO ensure that we only ever have one primary group. Probably need to change the model
+    primary_groups = indicator.groups[:2] # TODO not more than two primary groups. Probably need to change the model
```
Comment (kilemensi) [inline wazimap_ng/datasets/tasks/indicator_data_extraction.py:72]: QQ, in what scenarios would one need more than two groups?

File: wazimap_ng/datasets/tasks/indicator_data_extraction.py [hunk 3]
```diff
@@ -66,15 +69,14 @@ def add_subindicator(self, data_blob):
...
 
     models.IndicatorData.objects.filter(indicator=indicator).delete()
-    groups = ["data__" + i for i in indicator.dataset.groups]
```
Comment (kilemensi) [inline wazimap_ng/datasets/tasks/indicator_data_extraction.py:72]: Where was this used?

File: wazimap_ng/datasets/tasks/indicator_data_extraction.py [hunk 4]
```diff
@@ -83,9 +85,10 @@ def indicator_data_extraction(indicator, **kwargs):
 
                 counts = extract_counts(indicator, qs_subindicator)
                 sorter.add_data(group, subindicator, counts)
-        else:
-            counts = extract_counts(indicator, qs)
-            sorter.add_subindicator(counts)
+
+    qs = models.DatasetData.objects.filter(dataset=indicator.dataset, data__has_keys=primary_groups)
+    counts = extract_counts(indicator, qs)
+    sorter.add_subindicator(counts)
```
Comment (kilemensi) [inline wazimap_ng/datasets/tasks/indicator_data_extraction.py:91]: Shouldn't these two be part of `else:` above?

File: wazimap_ng/utils.py [hunk 5]
```diff
@@ -489,7 +489,7 @@ def rearrange(in_arrs, order):
     """
     out_arrs = []
     for arr in in_arrs:
-        out_arrs += [[arr[idx] for idx in order] + [arr[-1]]]
+        out_arrs += [[arr[idx] for idx in order if arr[idx] != arr[-1]] + [arr[-1]]]
```
Comment (kilemensi) [inline wazimap_ng/utils.py:492]: Can you explain this change a bit?

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: wazimap_ng/datasets/tasks/indicator_data_extraction.py (hunk 2 above)
Comment (KhadijaMahanga) [reply wazimap_ng/datasets/tasks/indicator_data_extraction.py:72]: @kilemensi  To my understanding, having more than 2 groups as `primary`  means we're moving away from 2 axis charts.  Say for-instance we have count (y-axis)- age, status, and year as primary groups..we can't  plot them all at once on our usual x-y axis. 

We can plot 
- count -> age & year .....while status becomes a filter group (not primary)
- count -> age & status.....while year becomes a filter group 
- count -> year & status ......while age becomes a filter group

File: wazimap_ng/datasets/tasks/indicator_data_extraction.py (hunk 3 above)
Comment (KhadijaMahanga) [reply wazimap_ng/datasets/tasks/indicator_data_extraction.py:72]: It was redundant, not used

File: wazimap_ng/utils.py (hunk 5 above)
Comment (KhadijaMahanga) [reply wazimap_ng/utils.py:492]: @kilemensi  The depth/length of indicator object now varies between 9 and 8, and this function re-arranges list except for the last element which stays in place (note: last element is usually the numeric value).

Here, I making sure that the element being moved is not the last element.

File: wazimap_ng/datasets/tasks/indicator_data_extraction.py (hunk 2 above)
Comment (kilemensi) [reply wazimap_ng/datasets/tasks/indicator_data_extraction.py:72]: Cool, so lets update the comment there with this info.

File: wazimap_ng/utils.py (hunk 5 above)
Comment (kilemensi) [reply wazimap_ng/utils.py:492]: Cool, cool... So we're sure the values of `arr` are all different, right?
